File: libJava/com/earndero/sqlwrapper/java/Cursor.java
package com.earndero.sqlwrapper.java;

import java.sql.SQLException;

public class Cursor {
    private final java.sql.ResultSet rs;
    Cursor(java.sql.ResultSet rs) {
        this.rs = rs;
    }

    public boolean next() {
        try {
            return rs.next();
        } catch (java.sql.SQLException throwables) {
            throwables.printStackTrace();
        }
        return false;
    }

    public void close() throws SQLException {
        rs.close();
    }

    public String getString(String name) {
        try {
            return rs.getString(name);
        } catch (java.sql.SQLException throwables) {
            throwables.printStackTrace();
        }
        return "";
    }

    public int getInt(String name) {
        try {
            return rs.getInt(name);
        } catch (java.sql.SQLException throwables) {
            throwables.printStackTrace();
        }
        return 0;
    }

    public double getFloat(String name) {
        try {
            return rs.getFloat(name);
        } catch (java.sql.SQLException throwables) {
            throwables.printStackTrace();
        }
        return 0;
    }
}

File: libJavaKt/com/earndero/sqlwrapper/javakt/Cursor.kt
package com.earndero.sqlwrapper.javakt

import java.sql.ResultSet
import kotlin.Throws
import java.sql.SQLException

class Cursor internal constructor(private val rs: ResultSet?) {
    operator fun next(): Boolean {
        try {
            return rs!!.next()
        } catch (throwables: SQLException) {
            throwables.printStackTrace()
        }
        return false
    }

    @Throws(SQLException::class)
    fun close() {
        rs!!.close()
    }

    fun getString(name: String?): String {
        try {
            return rs!!.getString(name)
        } catch (throwables: SQLException) {
            throwables.printStackTrace()
        }
        return ""
    }

    fun getInt(name: String?): Int {
        try {
            return rs!!.getInt(name)
        } catch (throwables: SQLException) {
            throwables.printStackTrace()
        }
        return 0
    }

    fun getFloat(name: String?): Double {
        try {
            return rs!!.getFloat(name).toDouble()
        } catch (throwables: SQLException) {
            throwables.printStackTrace()
        }
        return 0.0
    }
}